core/controllers/cron_test.py

```diff
@@ -447,6 +449,227 @@ def test_email_not_sent_if_reviewer_ids_is_empty(self) -> None:
+        # Give reviewer rights to review translations in the given language
+        # code.
+        user_services.allow_user_to_review_translation_in_language(
+            self.reviewer_id, self.language_code)
+        # Create a translation suggestion so that the reviewer has something
+        # to be notified about.
+        translation_suggestion = (
+            self._create_translation_suggestion_for_en_language())
+
+        self.expected_reviewable_suggestion_email_info = (
+            suggestion_services
+            .create_reviewable_suggestion_email_info_from_suggestion(
+                translation_suggestion))
+
+        self.can_send_emails = self.swap(feconf, 'CAN_SEND_EMAILS', True)
+        self.cannot_send_emails = self.swap(feconf, 'CAN_SEND_EMAILS', False)
+        self.testapp_swap = self.swap(
+            self, 'testapp', webtest.TestApp(main.app_without_context))
+
+    def test_email_not_sent_if_sending_emails_is_not_enabled(self) -> None:
+        # Here we use object because we need to spy on
+        # send_reviewer_notifications method and assert
+        # if it's being called.
+        with mock.patch.object(
+            email_manager, 'send_reviewer_notifications',
+            new_callable=mock.Mock
+        ) as mock_send:
+            self.login(
+                self.CURRICULUM_ADMIN_EMAIL, is_super_admin=True)
+            swap_platform_parameter_value = self.swap_to_always_return(
+                platform_feature_services,
+                'get_platform_parameter_value',
+                True
+            )
+
+            with self.cannot_send_emails, self.testapp_swap, swap_platform_parameter_value: # pylint: disable=line-too-long
+                self.get_json(
+                    '/cron/mail/reviewers/new_contr' +
+                    'ibutor_dashboard_suggestions')
+
+            mock_send.assert_not_called()
+            self.logout()
+
+    def test_email_sent_to_reviewer_if_sending_reviewer_emails_is_enabled(self) -> None: # pylint: disable=line-too-long
+        # Here we use object because we need to spy on
+        # send_reviewer_notifications method and assert
+        # if it's being called.
+        with mock.patch.object(
+            email_manager, 'send_reviewer_notifications',
+            new_callable=mock.Mock
+        ) as mock_send:
+            self.login(
+                self.CURRICULUM_ADMIN_EMAIL, is_super_admin=True)
+            swap_platform_parameter_value = self.swap_to_always_return(
+                platform_feature_services,
+                'get_platform_parameter_value',
+                True
+            )
+
+            with self.can_send_emails, self.testapp_swap, swap_platform_parameter_value: # pylint: disable=line-too-long
+                self.get_json(
+                    '/cron/mail/reviewers/new_cont' +
+                    'ributor_dashboard_suggestions')
+
+            for call_args in mock_send.call_args_list:
```

Comment:
Can you please check if you can use `assert_called_once_with`? Or if you assert the entire object at once? This will make for a more readable test. Same for everywhere else you are doing this.

Reply:
Done